dns/controller_test.go

```diff
@@ -0,0 +1,104 @@
+package dns
+
+import (
+	"testing"
+
+	"time"
+
+	"fmt"
+
+	"github.com/sky-uk/feed/k8s"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+const smallWaitTime = time.Millisecond * 50
+
+type fakeClient struct {
```

Comment:
Can we share the mock for the k8s client? It will be a pain to update both everytime we change it